src/components/application_manager/src/commands/button_notification_to_mobile.cc

```diff
@@ -0,0 +1,231 @@
+  SDL_LOG_AUTO_TRACE();
+  using namespace application_manager::strings;
+  return (*message_).keyExists(msg_params) &&
+         (*message_)[msg_params].keyExists(param_name);
+}
+
+void ButtonNotificationToMobile::HandleCustomButton(
+    app_mngr::ApplicationSharedPtr app) {
+  SDL_LOG_AUTO_TRACE();
+  using namespace application_manager::strings;
+
+  if (!app) {
+    SDL_LOG_ERROR("Application doesn't exist.");
+    return;
+  }
+
+  // app_id is mandatory for CUSTOM_BUTTON notification
+  if (!DoesParamExist(strings::app_id)) {
+    SDL_LOG_ERROR("CUSTOM_BUTTON mobile notification without app_id.");
+    return;
+  }
+
+  // custom_button_id is mandatory for CUSTOM_BUTTON notification
+  if (!DoesParamExist(hmi_response::custom_button_id)) {
+    SDL_LOG_ERROR(
+        "CUSTOM_BUTTON mobile notification without custom_button_id.");
+    return;
+  }
+
+  if (!app->IsSubscribedToButton(mobile_apis::ButtonName::CUSTOM_BUTTON)) {
+    SDL_LOG_ERROR("Application " << app->app_id()
+                                 << " is not subscribed on custom buttons");
+    return;
+  }
+
+  const uint32_t custom_btn_id =
+      (*message_)[msg_params][hmi_response::custom_button_id].asUInt();
+
+  if (!app->IsSubscribedToSoftButton(custom_btn_id)) {
+    SDL_LOG_ERROR("Application doesn't subscribed to this custom_button_id.");
+    return;
+  }
+
+  const auto window_id = app->GetSoftButtonWindowID(custom_btn_id);
+  (*message_)[strings::msg_params][strings::window_id] = window_id;
+
+  if (mobile_api::HMILevel::HMI_NONE == app->hmi_level(window_id)) {
+    SDL_LOG_WARN("CUSTOM_BUTTON mobile notification is allowed only "
+                 << "in FULL, LIMITED or BACKGROUND hmi level");
+    return;
+  }
+
+  SendButtonNotification(app);
+  return;
+}
+
+void ButtonNotificationToMobile::HandleOKButton(
+    app_mngr::ApplicationSharedPtr app) {
+  SDL_LOG_AUTO_TRACE();
+  using namespace application_manager;
+  SDL_LOG_DEBUG("OK button received");
+
+  if (app) {
+    const auto btn_id = static_cast<mobile_apis::ButtonName::eType>(
+        (*message_)[strings::msg_params][hmi_response::button_name].asInt());
+
+    if (app->IsSubscribedToButton(btn_id)) {
+      SendButtonNotification(app);
+    } else {
+      SDL_LOG_ERROR("Application " << app->app_id()
+                                   << " is not subscribed to button "
+                                   << btn_id);
+    }
+
+    return;
+  }
+
+  const auto subscribed_apps = SubscribedApps();
+  const auto app_ptr =
+      std::find_if(subscribed_apps.begin(),
+                   subscribed_apps.end(),
+                   [](const ApplicationSharedPtr subscribed_app) {
+                     return subscribed_app->IsFullscreen();
+                   });
+
+  if (app_ptr != subscribed_apps.end()) {
+    SDL_LOG_DEBUG("Sending button press for app in FULL with app id: "
+                  << (*app_ptr)->app_id());
+    SendButtonNotification(*app_ptr);
+    return;
```

Comment:
```suggestion
```

Reply:
@iCollin removed in https://github.com/smartdevicelink/sdl_core/pull/3743/commits/daa1060833441f38a7219d8a852c6d44e852e76e